treewide: mark failing packages as broken

mdaniels5757 opened the pull request:
On 12 separate Hydra evaluations (which I selected for their relatively low failure numbers, and which span over a year), these packages have done nothing other than fail for all platforms.
That is, if a package had a status other than "Failed" on any platform for any of the selected evaluations, it is not included here.
For Python packages and Linux kernel modules, they also fail across all built Python versions (so if the package succeeds with Python 3.12 but fails with Python 3.13, it's not included here).

The marking itself was done with `pkgs/common-updater/scripts/mark-broken`, generation of the packages to be marked was done with a modification to https://git.helsinki.tools/janne.hess/zhf/-/commits/master and a custom Python script.

This should have 0 rebuilds.

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] 0 rebuilds.
- Built on platform:
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [ ] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [X] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

mdaniels5757 committed:
treewide: mark failing packages as broken

On 16 separate Hydra evaluations (which I selected for their relatively
low failure numbers, and which span just over a year), these packages have done
nothing other than fail for all platforms.
For Python packages, they also fail across all built Python versions.

iedame commented:
cloudcompare was fixed by https://github.com/NixOS/nixpkgs/pull/454578
python3Packages.uqbar was fixed by https://github.com/NixOS/nixpkgs/pull/451495

SuperSandro2000 commented:
Can we add a short comment why those packages where marked broken? Otherwise it is not easy to assess afterwards if it is worth fixing them or it can be just dropped.

qbisi commented:
some packages failed due to cmake v4 error. e.g. itk_5_2/getdp. itk is not broken, only mark broken for itk version 5.2.

Also, getdp's original source url is not available anymore, the fix will be getdp 4.0 (currently 4.0 is not released).

mdaniels5757 commented on pkgs/development/mobile/androidenv/default.nix:
```diff
@@ -32,6 +32,21 @@ lib.recurseIntoAttrs rec {
   meta = {
+    # Marked broken 2025-11-01 because it has failed on Hydra for nearly a year.
+    broken = true;
+    # Marked broken 2025-11-01 because it has failed on Hydra for nearly a year.
```
Lol, it tried to mark it broken once for every subpackage that failed.

This file should be untouched now.

mdaniels5757 commented:
I've removed itk from this PR.

getdp is broken, and has been on 10 occasions spanning about a year. So it should be marked as such. If it's fixed, it can be unmarked as broken.

Ma27 commented on pkgs/development/php-packages/datadog_trace/default.nix:
```diff
@@ -55,6 +57,5 @@ buildPecl rec {
       bsd3
     ];
     teams = [ lib.teams.php ];
-    broken = lib.versionAtLeast php.version "8.4";
```
Did you check if this applies to all PHP versions?

Ma27 commented on pkgs/development/tools/database/indradb/default.nix:
```diff
@@ -15,6 +15,9 @@ let
   meta = with lib; {
+    # Marked broken 2025-11-01 because it has failed on Hydra for nearly a year.
+    broken = true;
+    # Marked broken 2025-11-01 because it has failed on Hydra for nearly a year.
```
This also looks wrong.

mdaniels5757 commented:
Fixed merge conflicts, updated with two recent evals ([1809932](https://hydra.nixos.org/eval/1809932) and [1820178](https://hydra.nixos.org/eval/1820178), and updated the date in the comments.

I only did `git reset HEAD^` and then manually `git restore`d files, so I could not have introduced new issues (except, I suppose, from find-replacing the comments).

bryanhonof commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 451447`
Commit: `b769c8a12924d82b4a16ee73b3a381ca34effb8d`

bryanhonof reviewed (approved):
Approved automatically following the successful run of `nixpkgs-review`.

mdaniels5757 commented:
Fixed merge conflicts, updated with two recent evals ([1820323](https://hydra.nixos.org/eval/1820323) and [1820232](https://hydra.nixos.org/eval/1820232), and updated the date in the comments.

Again, there were no changes except from the find-replace and removing a file that had a conflict, so I could not have introduced new issues (except, I suppose, from find-replacing the comments).

lukegb reviewed (approved):
Sounds good (at least from my perspective for oven-media-engine). Are you planning on marking these as broken in 25.11 as well?

mdaniels5757 commented:
Yes, I was going to backport this. Thanks for the reminder!

corngood commented on pkgs/os-specific/cygwin/w32api/default.nix:
```diff
@@ -32,6 +32,8 @@
     };
 
     meta = {
+      # Marked broken 2025-11-28 because it has failed on Hydra for at least one year.
```
This broke eval of everything on cygwin.  master already had a fix for this by limiting the platform to cygwin.  I'd like to remove the broken flag in #468157.

mdaniels5757 replied:
Noted; looks like 9dd0669e491ee8ce9064ab94a8301b189eabf075 post-dates my PR, which would explain why I missed it :).
